std-bits/table/src/main/java/org/enso/table/data/column/builder/Builder.java

```diff
@@ -53,17 +70,19 @@ static Builder getForType(StorageType type, int size, ProblemAggregator problemA
    *
    * @param size the initial size of the builder.
    */
-  static Builder getInferredBuilder(int size, ProblemAggregator problemAggregator) {
-    return new InferredBuilder(size, problemAggregator, false);
+  static Builder getInferredBuilder(long size, ProblemAggregator problemAggregator) {
+    checkSize(size);
+    return new InferredBuilder((int)size, problemAggregator, false);
```

Comment:
```suggestion
    int checkedSize = checkSize(size);
    return new InferredBuilder(checkedSize, problemAggregator, false);
```

`checkSize` guarantees that `size` can be safely cast to `int`. Perhaps it could return a 'proof' of this just by returning the `int` size? Then we can avoid an additional cast.

Having these casts makes the reader wonder if they are safe. Opting to "[parse instead of validating](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/)" makes it clear that there is no edge cases omitted here.

Reply:
thats only true for the current implementation, but as will be a while before we exceed `int` can do that.

Reply:
That was my reasoning - its a `private` helper method so if implementation changes in the future, it can be updated accordingly.